glean-core/src/lib.rs

```diff
@@ -201,12 +201,17 @@ impl Glean {
         let event_data_store = EventDatabase::new(&cfg.data_path)?;
 
         // Create an upload manager with rate limiting of 10 pings every 60 seconds.
-        let mut upload_manager =
-            PingUploadManager::new(&cfg.data_path, &cfg.language_binding_name, false);
+        let mut upload_manager = PingUploadManager::new(&cfg.data_path, &cfg.language_binding_name);
         upload_manager.set_rate_limiter(
             /* seconds per interval */ 60, /* max tasks per interval */ 15,
         );
 
+        // We only scan the pending ping sdirectories when calling this from a subprocess,
```

Comment:
Having this makes me question the name of this function. If you have better ideas I'd be up for changing it.